tmt/steps/provision/__init__.py

```diff
@@ -8,36 +8,57 @@
 from click import echo
 
 from tmt.utils import SpecificationError
+from tmt.steps.provision import vagrant, localhost
 
 
 class Provision(tmt.steps.Step):
     """ Provision step """
-
-    # Default implementation for provision is a virtual machine
+    name = 'provision'
+
+    # supported provisioners are not loaded automatically, import them and map them in how_map
+    how_map = {
+        'vagrant': vagrant.ProvisionVagrant,
+        'libvirt': vagrant.ProvisionVagrant,
+        'virtual': vagrant.ProvisionVagrant,
+        'local': localhost.ProvisionLocalhost,
+        'localhost': localhost.ProvisionLocalhost
+    }
+
+    # default provisioner
     how = 'virtual'
 
     def __init__(self, data, plan):
-        super(Provision, self).__init__(data, plan)
         # List of provisioned guests
         self.guests = []
+
+        # Parent
         self.super = super(Provision, self)
 
+        # Initialize parent
+        self.super.__init__(data, plan)
+
+    def _check_data(self):
+        """ Validate input data """
+
+        # if not specified, use 'virtual' provisioner as default
+        for data in self.data:
+            how = data['how']
+            # is how supported?
+            if how not in self.how_map:
+                raise tmt.utils.SpecificationError("How '{}' in plan '{}' is not implemented".format(how, self.plan))
```

Comment:
I think there's no way to forward-check whether vagrant supports it, so we should simply input all the non-localhost `how`s to Vagrant. Please see https://github.com/psss/tmt/pull/43#issuecomment-553502265.

Reply:
Agreed, ok. Will route everything other via vagrant.